modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/web/ModelController.java

```diff
@@ -43,25 +44,26 @@
         return modelService.getModels(metricKey);
     }
 
-    @ApiOperation(value = "Add model hyper params")
-    @PostMapping(value = "/addModelParams", produces = MediaType.APPLICATION_JSON_VALUE)
-    public String addModelParams(@RequestBody ModelParams modelParams) {
-        modelService.addModelParams(modelParams);
+    @ApiOperation(value = "Add hyperparams")
+    @PutMapping(value = "/models/{uuid}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public String addHyperparams(@PathVariable String uuid, @RequestBody Hyperparams hyperparams) {
+        modelService.addModelParams(uuid, hyperparams);
         return "Model params saved successfully";
     }
 
     @ApiOperation(value = "Mark model to rebuild")
-    @PutMapping(value = "/markToRebuild", produces = MediaType.APPLICATION_JSON_VALUE)
-    public String markToRebuild(@RequestBody RebuildParams rebuildParams) {
-        modelService.markToRebuild(rebuildParams.getModelUUID(), rebuildParams.getMetricKey(),
-                rebuildParams.getToRebuild());
+    @PutMapping(value = "/rebuildModel/{uuid}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public String markToRebuild(@PathVariable String uuid, @RequestBody RebuildParams rebuildParams) {
+        modelService.markToRebuild(uuid, rebuildParams.getMetricKey(),
+                rebuildParams.isToRebuild());
         return "Model marked for rebuild";
     }
 
+    @Deprecated
     @ApiOperation(value = "Update threshold for a given model")
-    @PutMapping(value = "/updateThreshold", produces = MediaType.APPLICATION_JSON_VALUE)
-    public String updateThresholds(@RequestBody ThresholdParams thresholdParams) {
-        modelService.updateThresholds(thresholdParams.getModelUUID(), thresholdParams.getMetricKey(),
+    @PutMapping(value = "/updateThreshold/{uuid}", produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
REST endpoints should be nouns. Try

`/models/:uuid/thresholds`

or something similar